secret-mode.el

```diff
@@ -27,15 +27,31 @@
 
 ;;; Code:
 
-(defconst secret-mode-table
-  (let ((disptbl (make-display-table)))
-    (dotimes (i 1024)
-      (aset disptbl i
-            (pcase (get-char-code-property i 'general-category)
-              ((or 'Cc 'Cf 'Zs 'Zl 'Zp) nil)
-              ('Lu (vector (make-glyph-code ?▆)))
-              (_   (vector (make-glyph-code ?▃))))))
-    disptbl)
+(eval-and-compile
+  (defconst secret-mode--max-letter-char-code
+    (eval-when-compile
+      (let ((max-code-point #x10FFFF)    ; Max Unicode code point
+            (letter-categories '(Lu Ll)) ; Upper and Lower case letters
+            result)
+        (dotimes (i (1+ max-code-point))
+          (when (memq (get-char-code-property i 'general-category)
+                      letter-categories)
+            (setq result i)))
+        result))
+    "Maximum Unicode code point that designates a letter.")
```

Comment:
That loop is indeed expensive. That's why It is only computed once, when the package is complied (by default Emacs compiles the package once, during the package installation).

Reply:
Because the expensive loops are being executed only once, during the package installation, I see no reason to use hardcoded magic number, instead of a more general solution.